champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/passives/SmokeBomb.java

```diff
@@ -10,162 +10,62 @@
 import me.mykindos.betterpvp.core.components.champions.SkillType;
 import me.mykindos.betterpvp.core.effects.Effect;
 import me.mykindos.betterpvp.core.effects.EffectType;
-import me.mykindos.betterpvp.core.effects.events.EffectClearEvent;
 import me.mykindos.betterpvp.core.effects.events.EffectExpireEvent;
 import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
 import me.mykindos.betterpvp.core.listener.BPvPListener;
 import me.mykindos.betterpvp.core.utilities.UtilMessage;
 import me.mykindos.betterpvp.core.utilities.UtilPlayer;
 import me.mykindos.betterpvp.core.utilities.UtilServer;
+import me.mykindos.betterpvp.core.utilities.UtilTime;
 import org.bukkit.Particle;
 import org.bukkit.Sound;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
-import org.bukkit.event.block.Action;
-import org.bukkit.event.block.BlockBreakEvent;
-import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.PlayerAttemptPickupItemEvent;
 import org.bukkit.event.player.PlayerInteractEvent;
-import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.potion.PotionEffect;
 import org.bukkit.potion.PotionEffectType;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import java.util.Iterator;
-import java.util.Map.Entry;
-import java.util.Optional;
 import java.util.WeakHashMap;
 
 @Singleton
 @BPvPListener
 public class SmokeBomb extends Skill implements ToggleSkill, CooldownSkill, Listener {
 
+    private final WeakHashMap<Player, Long> smoked = new WeakHashMap<>();
```

Comment:
I still don't like using players in a weakhashmap, you can't listen to when an object is gc'd and it may prevent clean up in other places, im also gonna go ahead and assume this is more overhead compared to a standard hashmap

Reply:
I don't see the first two problems as an issue because in the case of skills, it's not really important to us if we know when an object is GC'd or not, and weak maps store only weak references to its keys so it won't prevent cleanup elsewhere. I don't really know if it's less performant and resource-consuming but I would assume it's better than us having to manually clean entries up by calling `Bukkit.getPlayer` to check if they're still online.

I believe the getPlayer variation with a UUID parameter is a constant execution time as well so I'll go ahead and change the map to use UUID keys. We could probably do this for most skills as well.